.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,87 @@
+name: Create Release PR
+on:
+  workflow_dispatch:
+  schedule:
+    # Midnight on the 1st of every 2nd month
+    - cron: 0 0 1 */2 *
+jobs:
+  create-release-pr:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Increment Patch Version
+        id: version
+        run: |
+          prefix=$(grep -Po '(?<=HAIL_MAJOR_MINOR_VERSION := )(\d+\.\d+)' hail/Makefile)
+          patch=$(grep -Po '(?<=HAIL_PATCH_VERSION := )(\d+)' hail/Makefile)
+          
+          n=$((patch + 1))
+          
+          sed -Ei 's|(HAIL_PATCH_VERSION := )[[:digit:]]+|\1'"${n}"'|' hail/Makefile
+          sed -Ei 's|(hailPatchVersion = ")[[:digit:]]+|\1'"${n}"'|' hail/build.mill
+          
+          echo "old=${prefix}.${patch}" >> "${GITHUB_OUTPUT}"
+          echo "new=${prefix}.${n}" >> "${GITHUB_OUTPUT}"
+          echo "next=${prefix}.$((n + 1))" >> "{GITHUB_OUTPUT}"
+
+      - name: Generate Changelog Entries
+        run: |
+          git log "${{ steps.version.outputs.old }}"..HEAD --format='%s' |\
+            grep -P '\[all|(com(biner|piler)|dep(s|endencies)|hail(?!ctl|genetics|jwt|top)|ir|jvm|p(ip|ython)|q(uery|ob)|vds)[^\]]*\]' |\
+            sed -E 's/\[[^\]+\] (.*) \(#([[:digit:]]*)\)/- (hail#\2) \1/' |\
+            sed 1i"## Version ${{ steps.version.outputs.new }}\n" |\
+            sed -e '$a\'$'\n' > HAIL_ENTRY
+          
+          sed -i '/## Version '"${{ steps.version.outputs.old }}"'/e cat HAIL_ENTRY' \
+            hail/python/hail/docs/change_log.md
+          
+          git log "${{ steps.version.outputs.old }}"..HEAD --format='%s' |\
+            grep -P '\[(a(ll|uth)|batch|dataproc|fs|hail(ctl|genetics|top)|infra|services)[^\]]*\]' |\
+            sed -E 's$\[[^\]+\] (.*) \(#([[:digit:]]*)\)$- (\`#\2 <https://github.com/hail-is/hail/pulls/\2>\`__) \1$' |\
+            sed 1i"**Version ${{ steps.version.outputs.new }}**\n" |\
+            sed -e '$a\'$'\n' > HAILTOP_ENTRY
+          
+          sed -i -e '/Change Log/{N; N; r HAILTOP_ENTRY' -e '}' \
+            hail/python/hailtop/batch/docs/change_log.rst
+
+      - name: Commit and Push
+        run: |
+          git config --global user.name "${GITHUB_ACTOR}"
```

Comment:
This is apparently "the last user to schedule or trigger the workflow", which presumably will be you. I think by default actions in a repo will be authed with a default "github" user with permissions within that repo, so I'm not sure this - or the config setting below - is actually necessary

Reply:
I see. Thanks for this. These things are a pain to discover and test before they get merged.

Reply:
Turns out it's needed https://github.com/hail-is/hail/actions/runs/14768079965/job/41463259374

Reply:
https://github.com/hail-is/hail/pull/14874